src/helper/cookie/index.ts

```diff
@@ -47,8 +71,27 @@ export const getSignedCookie: GetSignedCookie = async (c, secret, key?) => {
 }
 
 export const setCookie = (c: Context, name: string, value: string, opt?: CookieOptions): void => {
-  const cookie = serialize(name, value, { path: '/', ...opt })
-  c.header('set-cookie', cookie, { append: true })
+  // Cookie names prefixed with __Secure- can be used only if they are set with the secure attribute.
+  // Cookie names prefixed with __Host- can be used only if they are set with the secure attribute, must have a path of / (meaning any path at the host)
+  // and must not have a Domain attribute.
+  // Read more at https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#cookie_prefixes'
+  const cookies = []
+  if (opt?.securePrefix) {
+    cookies.push(serialize('__Secure-' + name, value, { path: '/', secure: true, ...opt }))
```

Comment:
Do not change the secure attribute.

```suggestion
    cookies.push(serialize('__Secure-' + name, value, { path: '/', ...opt, secure: true  }))
```